graphqlmetrics/core/metrics_service.go

```diff
@@ -34,55 +32,62 @@ type SchemaUsageRequestItem struct {
 	Claims      *utils.GraphAPITokenClaims
 }
 
+type ProcessorConfig struct {
+	Interval     time.Duration
+	MaxWorkers   int
+	MaxBatchSize int
+	MaxQueueSize int
+}
+
 type MetricsService struct {
 	logger *zap.Logger
 
 	// conn is the clickhouse connection
 	conn clickhouse.Conn
 
 	// opGuardCache is used to prevent duplicate writes of the same operation
-	opGuardCache *lru.Cache[string, struct{}]
-
-	processor *batch.Processor[SchemaUsageRequestItem]
+	opGuardCache *ristretto.Cache[string, struct{}]
 
-	pool *pond.WorkerPool
+	processor *batchprocessor.BatchProcessor[SchemaUsageRequestItem]
 }
 
 // NewMetricsService creates a new metrics service
-func NewMetricsService(ctx context.Context, logger *zap.Logger, chConn clickhouse.Conn, config batch.ProcessorConfig) *MetricsService {
-	c, err := lru.New[string, struct{}](25000)
+func NewMetricsService(ctx context.Context, logger *zap.Logger, chConn clickhouse.Conn, processorConfig ProcessorConfig) *MetricsService {
+	cacheConfig := &ristretto.Config[string, struct{}]{
+		MaxCost:     50_000,
+		NumCounters: 50_000 * 10,
```

Comment:
I don't think I understand why it's written like this

Reply:
Readability